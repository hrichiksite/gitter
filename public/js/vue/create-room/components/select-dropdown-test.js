const mount = require('../../__test__/vuex-mount');
const { default: SelectDropdown } = require('./select-dropdown.vue');

describe('SelectDropdown', () => {
  it('matches snapshot', () => {
    const { wrapper } = mount(SelectDropdown, {
      items: [
        {
          id: 1
        },
        {
          id: 2
        },
        {
          id: 3
        }
      ],
      filterText: '',
      filterPlaceholder: 'Filter items...'
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Inputting new filter text emits event', () => {
    const { wrapper } = mount(SelectDropdown, {
      items: [],
      filterText: '',
      filterPlaceholder: 'Filter items...'
    });

    wrapper.find({ ref: 'filterTextInput' }).element.value = 'lookingforyou';
    wrapper.find({ ref: 'filterTextInput' }).trigger('input');

    expect(wrapper.emitted().filterInput).toBeTruthy();
  });
});
