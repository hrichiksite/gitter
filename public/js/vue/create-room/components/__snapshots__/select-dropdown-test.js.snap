// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectDropdown matches snapshot 1`] = `
<b-dropdown-stub
  boundary="scrollParent"
  class="select-dropdown"
  menuclass="select-dropdown-menu"
  offset="0"
  role="menu"
  splitbuttontype="button"
  text=""
  toggleclass="select-dropdown-button"
  toggletag="button"
  toggletext="Toggle Dropdown"
  variant="secondary"
>
   
  <b-dropdown-form-stub>
    <input
      class="filter-input"
      placeholder="Filter items..."
    />
  </b-dropdown-form-stub>
   
  <!---->
   
  <!---->
   
  <ul
    class="select-dropdown-item-list"
  >
    <b-dropdown-item-button-stub
      activeclass="active"
      buttonclass="[object Object]"
    />
    <b-dropdown-item-button-stub
      activeclass="active"
      buttonclass="[object Object]"
    />
    <b-dropdown-item-button-stub
      activeclass="active"
      buttonclass="[object Object]"
    />
  </ul>
</b-dropdown-stub>
`;
